sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -261,6 +262,78 @@ protected Symbol visitFunctionCall(FunctionCall node, ExpressionAnalysisContext
         }
 
         @Override
+        protected Symbol visitSimpleCaseExpression(SimpleCaseExpression node, ExpressionAnalysisContext context) {
+            return createCaseFunction(node.getWhenClauses(), node.getOperand(), node.getDefaultValue(), context);
+        }
+
+        @Override
+        protected Symbol visitSearchedCaseExpression(SearchedCaseExpression node, ExpressionAnalysisContext context) {
+            return createCaseFunction(node.getWhenClauses(), null, node.getDefaultValue(), context);
+        }
+
+        private Symbol createCaseFunction(List<WhenClause> whenClauseExpressions,
+                                          @Nullable Expression operandExpression,
+                                          @Nullable Expression defaultValue,
+                                          ExpressionAnalysisContext context) {
+            List<Symbol> operands = new ArrayList<>(whenClauseExpressions.size());
+            List<Symbol> results = new ArrayList<>(whenClauseExpressions.size());
+            Set<DataType> resultsTypes = new HashSet<>(whenClauseExpressions.size());
+
+            // check for global operand
+            Symbol operandLeftSymbol = null;
+            if (operandExpression != null) {
+                operandLeftSymbol = operandExpression.accept(innerAnalyzer, context);
+            }
+
+            for (WhenClause whenClause : whenClauseExpressions) {
+                Symbol operandRightSymbol = whenClause.getOperand().accept(innerAnalyzer, context);
+                Symbol operandSymbol = operandRightSymbol;
+
+                if (operandLeftSymbol != null) {
+                    FunctionIdent ident = new FunctionIdent(EqOperator.NAME,
```

Comment:
Would it make sense to add some helper functions to EqOperator? 
There is already an existing EqOperator.createInfo method which could probably be exposed?